talpid-wireguard/src/lib.rs

```diff
@@ -144,7 +144,7 @@ impl Error {
 pub struct WireguardMonitor {
     runtime: tokio::runtime::Handle,
     /// Tunnel implementation
-    tunnel: Arc<Mutex<Option<Box<dyn Tunnel>>>>,
+    tunnel: Arc<AsyncMutex<Option<Box<dyn Tunnel>>>>,
```

Comment:
Why is a tokio mutex needed here?

Reply:
Due to a small refactor in another part of the code, I needed to be able to hold the lock across an await point. We need a tokio mutex for that. Unfortunately that change kinda cascaded all the way up